### ISSUE TITLE
Integration with Guetzli by Google

### ISSUE THREAD

**Issue by saurabheights** [danielgtaylor/jpeg-archive#56]
More info here at the [official repo](https://github.com/google/guetzli) and the [blog](https://opensource.googleblog.com/2017/03/guetzli-new-open-source-jpeg-encoder.html?m=1).

**Comment by danielgtaylor** [comment]
I was just about to open the same issue to investigate :smile:

Seems they require 300mb per megapixel of image, which is incredibly high. The project is also written in C++ which will make it more difficult to integrate, but it should be doable. I don't have a ton of time right now but here are some things to do:

- Figure out actual memory use for a typical photo. Can this be run on an average computer? What about using the script which runs multiple compressions in parallel?
- Get some statistics on the sort of improvements we can expect and the tradeoffs in memory and speed that they require. Should be able to use `jpeg-recompress` to get an idea of acceptable JPEG quality levels and then use the `guetzli` binary to compare output and speed to get a rough idea.
- Integrate both Guetzli and Butteraugli into the build process in a cross-platform fashion. They are using an interesting build system that might be useful for us as well.
- Add Butteraugli as an alternative to SSIM for image quality comparisons.
- Add Guetzli as an alternative to mozjpeg for the encoder.

**Comment by saurabheights** [comment]
Yes @danielgtaylor, let me be clear that guetzli is not ready for production. It takes good amount of time, coupled with use of 5GB memory for 24MP memory(I just commented this [here](https://github.com/mozilla/mozjpeg/issues/242)) makes it unusable for 32-bit systems. It's also too slow for people who process thousands of images a day. Just imagine the cost of number of servers to use. But that doesn't mean it's not useful for all. 

However, I opened this issue as this project is just born and will mature to address these issues. Also, please link the butteraugli enhancement issue to this.

I am in same position(don't have enough time) but in case if anyone picks any requirements mentioned above, do discuss before so as to not stop on other toes/work. I will try to pick one or other requirements and will address my results here. Of course for any manual image compression analysis, go ahead and do it. Since the more manual comparison is done the better as it is subjective. Thank you.

**Comment by vannitotaro** [comment]
Relevant: https://opensource.googleblog.com/2024/04/introducing-jpegli-new-jpeg-coding-library.html
